### DISCUSSION TITLE
Do you see a possible impact on the user-agent parsing with respect to Chrome's reduced user agent trials and rollout?

### DISCUSSION THREAD

**Discussion by JensRichnow** [nielsbasjes/yauaa#465] [Ideas]
Chrome announced for a while the [reduction of user-agent string](https://blog.chromium.org/2021/09/user-agent-reduction-origin-trial-and-dates.html) (and addition of respective client hints for finer-granular data). By the look of it the dropping of minor versioning (zeroing out) etc this shouldn't impact the user-agent parsing?

**Comment by nielsbasjes** [comment]
Yes, this manipulation of the useragent has quite a big impact on the parsing and analysis.
Mainly because there are now several situations where specific fields are now known to be nonsense.

Luckily [I have been aware](https://github.com/nielsbasjes/yauaa/issues/193) of these plans for about 2 years and I have been closely following what Chromium (i.e. Chrome, Edge, etc.) and Firefox have been doing in this area.

The current state is that Yauaa 6.9 already handles all documented planned variants of freezing the Chromium/Chrome/Edge/... UserAgent and manipulation that I'm aware of. https://www.chromium.org/updates/ua-reduction/

See this file for the analysis results for each of these cases https://github.com/nielsbasjes/yauaa/blob/main/analyzer/src/main/resources/UserAgents/GoogleChromeFrozen.yaml

In addition it handles cases like
- Report frozen (=manipulated) Mac OS X versions as `??`:
  - Always [10_15_7](https://bugs.chromium.org/p/chromium/issues/detail?id=1175225) since Chrome 90.
  - Always [10.15](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent/Firefox#macintosh) since Firefox 87.
- Report frozen `Windows 10` on Firefox 88+ as `Windows >=10`

If you are aware of any new manipulations that are documented I would really appreciate an issue with the link to such documentation.

**Reply by JensRichnow** [reply]
Many thanks for your fast reply! There is one thing I wanted to clarify further: I run the [Chrome provided example UA](https://www.chromium.org/updates/ua-reduction/#sample-ua-strings-phase-4)s for the various phases through the parser and noticed that - depending on the type, ie, desktop, mobile or tablet - we don't get the OS version, device name and device brand anymore for phases 5/6. Is this the point when implementing the specific client hints are required?

**Reply by nielsbasjes** [reply]
In phases 5/6 the reduced useragents on Windows will always have `Windows NT 10.0; Win64; x64`, now this seems `Windows 10` but the reality is that this same value will be shown when running on `Windows 7` or `Windows 11`. That is why the result shows `Windows NT ??`: It is a modern Windows version but we cannot know which version exactly. Similar for Mac OS which is always `Macintosh; Intel Mac OS X 10_15_7`.

So in the near future the useragent will no longer have the information needed about the Operating System version. The main limitation is that this library only analyzes the UserAgent string and thus can only report on what is provided.

So yes, if you need something like the Operating System version (i.e. Windows 7/8/10/11) you will need a different source to get that.
The Client Hints seem to be that source for now.

Now I have done some experimenting myself with these headers and what I see right now is that the initial CH headers sent by a browser to a site only have the OS name, Browser+version, and if it is a mobile device; essentially the same as what is in the reduced useragent.
Only on the second request does my browser send the additional headers that were requested by the website (like the OS version).

I do not know what these Client Hints wil show once the Phase 5/6 are there, they may be reduced in a similar way.

https://github.com/WICG/ua-client-hints
